contrib/inventory/apstra_aos.py

```diff
@@ -0,0 +1,395 @@
+                    self.add_var_to_host(device.name, key, value)
+
+                    if key == 'os_family':
+                        self.add_host_to_group(value, device.name)
+                    elif key == 'hw_model':
+                        self.add_host_to_group(value, device.name)
+
+            ## Check if device is associated with a blueprint
+            ##   if it's create a new group
+            if 'blueprint_active' in device.value['status'].keys():
+                if 'blueprint_id' in device.value['status'].keys():
+                    bp = aos.Blueprints.find(method='id', key=device.value['status']['blueprint_id'])
+
+                    if bp:
+                        self.add_host_to_group(bp['display_name'], device.name)
+
+        #----------------------------------------------------
+        # Convert the inventory and return a JSON String
+        #----------------------------------------------------
+        data_to_print = ""
+        data_to_print += self.json_format_dict(self.inventory, True)
+
+        print(data_to_print)
+
+    def read_settings(self):
+        """ Reads the settings from the apstra_aos.ini file """
+
+        config = ConfigParser.SafeConfigParser()
+        config.read(os.path.dirname(os.path.realpath(__file__)) + '/apstra_aos.ini')
+
+        self.aos_server = config.get('aos', 'aos_server')
+        self.aos_server_port = config.get('aos', 'port')
+        self.aos_username = config.get('aos', 'username')
+        self.aos_password = config.get('aos', 'password')
+
+    def parse_cli_args(self):
+        """ Command line argument processing """
+
+        parser = argparse.ArgumentParser(description='Produce an Ansible Inventory file based on Apstra AOS')
+        parser.add_argument('--list', action='store_true', default=True, help='List instances (default: True)')
+        parser.add_argument('--host', action='store', help='Get all the variables about a specific instance')
+        self.args = parser.parse_args()
+
+    def json_format_dict(self, data, pretty=False):
+        """ Converts a dict to a JSON object and dumps it as a formatted string """
+
+        if pretty:
+            return json.dumps(data, sort_keys=True, indent=2)
+        else:
+            return json.dumps(data)
+
+    def add_host_to_group(self, group, host):
+
+        # Check if the group exist, if not initialize it
+        if group not in self.inventory.keys():
+            self.inventory[group] = {}
+            self.inventory[group]['hosts'] = []
+            self.inventory[group]['vars'] = {}
+
+        self.inventory[group]['hosts'].append(host)
+
+    def add_var_to_host(self, host, var, value):
+
+        # Check if the host exist, if not initialize it
+        if host not in self.inventory['_meta']['hostvars'].keys():
+            self.inventory['_meta']['hostvars'][host] = {}
+
+        self.inventory['_meta']['hostvars'][host][var] = value
+
+AosInventory()
```

Comment:
I'd recommend adding a main() and a 'if __name__ == __main__: main()' here, but thats optional.